### PR TITLE
Don't update exporter position to smaller value

### LINKED ISSUE #5294: Failed to take snapshot. Expected to find an indexed entry for given snapshot position, but found no matching indexed entry which contains this position
**Describe the bug**

After a specific time, no partitions were able to take a snapshot. Interesting things that happened at this time 
* 2 nodes were restarted
* Elastic was restarted and remained unavailable after this time. 

However not exporting should not affect finding the index for a snapshot position. 

```
I 2020-09-03T08:50:42.909800Z Finished taking snapshot, need to wait until last written event position 92177208 is committed, current commit position is 92177208. After that snapshot can be marked as valid. 
I 2020-09-03T08:50:43.025026Z Current commit position 92177208 is greater then 92177208, snapshot 72694299-5-1599123042546-92177202-92176016 is valid and has been persisted. 
I 2020-09-03T08:50:46.586284Z Finished taking snapshot, need to wait until last written event position 92834088 is committed, current commit position is 92834088. After that snapshot can be marked as valid. 
I 2020-09-03T08:50:46.678231Z Current commit position 92834088 is greater then 92834088, snapshot 73295199-6-1599123046239-92834086-92832638 is valid and has been persisted. 
W 2020-09-03T08:57:09.727479Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 92972134, but found no matching indexed entry which contains this position. 
W 2020-09-03T08:57:50.236950Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 92408747, but found no matching indexed entry which contains this position. 
W 2020-09-03T08:59:04.398724Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 96063780, but found no matching indexed entry which contains this position. 
W 2020-09-03T09:02:09.735510Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 93291699, but found no matching indexed entry which contains this position. 
W 2020-09-03T09:02:50.265595Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 92740810, but found no matching indexed entry which contains this position. 
W 2020-09-03T09:04:04.414313Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 96401054, but found no matching indexed entry which contains this position. 
W 2020-09-03T09:07:09.736556Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 93620744, but found no matching indexed entry which contains this position. 
```

Prior to first 'Failed to take snapshot', two node were restarted including elastic. But Elastic did not start and was not available after that. Following is logs from one broker. It cannot export because elastic is not available.

```
I 2020-09-03T08:56:10.274085Z Exporter opened 
I 2020-09-03T08:56:10.844931Z Processor starts reprocessing, until last source event position 92927359 
E 2020-09-03T08:56:11.278363Z Error on exporting record with key 4503599666512179 
I 2020-09-03T08:56:29.944010Z Processor finished reprocessing at event position 92927359 
...
...
W 2020-09-03T08:57:09.727479Z Failed to take snapshot. Expected to find an indexed entry for given snapshot position 92972134, but found no matching indexed entry which contains this position. 
W 2020-09-03T08:57:10.103332Z RaftServer{raft-partition-partition-3} - AppendRequest{term=5, leader=2, prevLogIndex=75643079, prevLogTerm=5, entries=0, commitIndex=75667861} to 0 failed: RaftError{type=UNAVAILABLE, message=null} 
```

[Link to logs](https://console.cloud.google.com/logs/viewer?interval=NO_LIMIT&project=zeebe-io&minLogLevel=0&expandAll=false&timestamp=2020-09-04T08%3A17%3A02.619000000Z&customFacets&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.container_name%3D%22zeebe-cluster%22%0Aresource.labels.pod_name%3D%22dd-medium-load-zeebe-2%22%0Aresource.labels.namespace_name%3D%22dd-medium-load%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Atimestamp%3D%222020-09-03T08%3A57%3A09.727479000Z%22%0AinsertId%3D%22abfy14t63d7mg2n2l%22&scrollTimestamp=2020-09-03T08%3A57%3A11.409581000Z)

**Environment:**
 - Zeebe Version: 0.25.SNAPSHOT

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

See https://github.com/zeebe-io/zeebe/issues/5294#issuecomment-702563848

>If we have exported all records, do a snapshot and restart the broker then there is nothing the export. The `ElasticSearchExporter` does an periodically flush, which also updates the lastExportedPosition, see [here](https://github.com/zeebe-io/zeebe/blob/develop/exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporter.java#L126). If the elasticsearch exporter hasn't seen any records after restart it will update the position to `-1`, since this is the default value.
>
>In the ExporterDirector we don't validate the position, whether it is smaller then the previous value which means we just update the position, see [here](https://github.com/zeebe-io/zeebe/blob/develop/broker/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java#L473).

## Related issues

closes https://github.com/zeebe-io/zeebe/issues/5294

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [X] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java [hunk 1]
```diff
@@ -463,7 +463,7 @@ private void updatePositionOnSkipIfUpToDate(final long eventPosition) {
     }
 
     private void updateExporterLastExportedRecordPosition(final long eventPosition) {
-      state.setPosition(getId(), eventPosition);
+      state.setPositionIfGreater(getId(), eventPosition);
       metrics.setLastUpdatedExportedPosition(getId(), eventPosition);
       position = eventPosition;
```
Comment (deepthidevaki) [inline broker/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java:468]: These should be updated only if `eventPosition` is greater than the current.

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
LGTM. :+1: Some small comments only.

File: broker/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java (hunk 1 above)
Comment (Zelldon) [reply broker/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java:468]: I have totally overseen that thanks. This reminds me that the export director is currently really hard to test. Maybe I can change that.

**Comment by Zelldon** [comment]
@deepthidevaki I found a way to test this better but this means a bit of refactoring. What do you think should I add this to this PR or do you want me to create a separate one for that?

**Comment by deepthidevaki** [comment]
You can add this to this PR.

**Comment by Zelldon** [comment]
@deepthidevaki I updated the PR please have a look

**Review by deepthidevaki — APPROVED** [review]
Nice :+1:

**Pull request merged**